server/core/runtime/plan_step_runner.go

```diff
@@ -165,10 +165,14 @@ func (p *PlanStepRunner) buildPlanCmd(ctx models.ProjectCommandContext, extraArg
 		envFileArgs = []string{"-var-file", envFile}
 	}
 
+	planMode := "-refresh"
+	if ctx.DestroyPlan {
+		planMode = "-destroy"
```

Comment:
could we use the decorator pattern here? Some details about why and how are here:
https://docs.lyft.net/eng/atlantis/internal/development/contributing.html#contributing-to-lyft-atlantis

I think we should be able to add to the extraArgs right?

Reply:
Added `DestroyPlanStepRunnerWrapper` decorator that appends destroy mode flag onto extra args in terraform plan command.